### PR TITLE
Make `data.py` ready for classification task

### PR DESCRIPTION
**Pull request by markus-hinsche**
**Changes:**
- Only add to path_cols if it is a path-like object

### CODE REVIEW

**Comment by kbressem** [comment]
I don't think this will work. What if label is a string, e.g. ["active changes", "structural changes"].

This would also be a great use case for try/except/else

**Comment by markus-hinsche** [comment]
Hmm, you are right. This is not good enough yet. 
`path.exists()` could check if the path exists on the file system. 
Any other ideas?

**Comment by kbressem** [comment]
Yes. I also thing path exists is needed. Also a nice check for correct paths. Double functionality with just one feature. Awesome 😎

File: trainlib/data.py [hunk 1]
```diff
@@ -199,7 +200,18 @@ def segmentation_dataloaders(
         train_df = train_df.sample(10)
         valid_df = valid_df.sample(5)
 
-    for col in image_cols + label_cols:
+    # Assemble columns that contain a path
+    path_cols = image_cols
+    for col in label_cols:
+        try:
+            path = train_df[col][0]
+            path = Path(path)
```
Comment (markus-hinsche) [inline trainlib/data.py:208]:
```suggestion
            path = Path(path)
            if not path.exits():
                continue
```

**Comment by markus-hinsche** [comment]
But it only checks the path at index 0, not all paths

**Comment by kbressem** [comment]
> But it only checks the path at index 0, not all paths

True, but this would catch errors with data_dir. 
Otherwise check of it ends with `\.[A-Za-z0-9]+`. Still not 100% fool-proof.

**Comment by kbressem** [comment]
Have a look at #68. It's fixed now. 

Sorry that I've ignored your changes but there were some deeper errors in the viewer classes which needed to be addressed.

**Review by kbressem — APPROVED** [review]
LGTM. But can you address the mypy error? Maybe use `Path` not `PurePath`?

File: trainlib/data.py [hunk 2]
```diff
@@ -200,7 +200,18 @@ def segmentation_dataloaders(
...
+        except (KeyError, TypeError):
+            continue
+        if isinstance(path, Path) and path.exits():
```
Comment (markus-hinsche) [inline trainlib/data.py:211]:
```suggestion
        if isinstance(path, Path) and path.exists():
```

**Comment by markus-hinsche** [comment]
Fixed mypy but now unit tests are failing. @kbressem Any idea why?

File: tests/test_trainer.py [hunk 3]
```diff
@@ -11,9 +11,18 @@ class TestSegmentationTrainer(unittest.TestCase):
...
+        except FileNotFoundError:
+            pass
+        try:
+            shutil.rmtree(self.config.model_dir)
+        except FileNotFoundError:
+            pass
+        try:
+            shutil.rmtree(self.config.data.cache_dir)
+        except FileNotFoundError:
+            pass
```
Comment (kbressem) [inline tests/test_trainer.py:25]:
```suggestion
        shutil.rmtree(self.config.run_id.split("/")[0], ignore_errors=True)
        shutil.rmtree(self.config.model_dir, ignore_errors=True)
        shutil.rmtree(self.config.data.cache_dir, ignore_errors=True)
        ```

**Review by kbressem — CHANGES_REQUESTED** [review]

File: tests/test_trainer.py [hunk 4]
```diff
@@ -11,9 +11,10 @@ class TestSegmentationTrainer(unittest.TestCase):
     config = TEST_CONFIG
 
     def tearDown(self) -> None:
-        shutil.rmtree(self.config.run_id.split("/")[0])
-        shutil.rmtree(self.config.model_dir)
-        shutil.rmtree(self.config.data.cache_dir)
+        shutil.rmtree(self.config.run_id.split("/")[0], ignore_errors=True)
+        shutil.rmtree(self.config.model_dir, ignore_errors=True)
+        shutil.rmtree(self.config.data.cache_dir, ignore_errors=True)
+        ```
```
Comment (kbressem) [inline tests/test_trainer.py:17]:
```suggestion

```

**Comment by kbressem** [comment]
> Fixed mypy but now unit tests are failing. @kbressem Any idea why?

`path_cols` was just a pointer to `image_cols`. So each change to `path_cols` also changes `image_cols`. 

Fixed that. Also made some minor changes on the way

**Review by kbressem — APPROVED** [review]
LGTM now

**Comment by markus-hinsche** [comment]
Great thanks!
